orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -139,34 +128,95 @@ private void verifyOtherServerGroupsAreTakingTraffic(String serverGroupName, Mon
       .filter(tsg -> location.equals(tsg.getLocation()))
       .collect(Collectors.toList());
 
-    Map<String, Integer> capacityByServerGroupName = targetServerGroups
-      .stream()
+    TargetServerGroup serverGroupGoingAway = targetServerGroups.stream()
+      .filter(sg -> serverGroupName.equals(sg.getName()))
+      .findFirst()
+      .orElseThrow(() -> new TrafficGuardException(format("Could not find server group '%s' in %s/%s with traffic guard configured.",
+        serverGroupName, account, location.getValue())));
+
+    verifyTrafficRemoval(serverGroupGoingAway, targetServerGroups, account, operationDescriptor);
+  }
+
+  public void verifyTrafficRemoval(TargetServerGroup serverGroupGoingAway,
+                                   Collection<TargetServerGroup> currentServerGroups,
+                                   String account,
+                                   String operationDescriptor) {
+    verifyTrafficRemoval(Collections.singletonList(serverGroupGoingAway),
+      currentServerGroups, account, operationDescriptor);
+  }
+
+  // if you disable serverGroup, are there other enabled server groups in the same cluster and location?
+  // TODO rz - Expose traffic guards endpoint in clouddriver
+  public void verifyTrafficRemoval(Collection<TargetServerGroup> serverGroupsGoingAway,
+                                   Collection<TargetServerGroup> currentServerGroups,
+                                   String account,
+                                   String operationDescriptor) {
+    Preconditions.checkArg(!serverGroupsGoingAway.isEmpty(), "serverGroupsGoingAway must not be empty");
+    Preconditions.checkArg(!currentServerGroups.isEmpty(), "currentServerGroups must not be empty");
```

Comment:
Do these bubble up as user facing error messages? If so, we should provide a better message. If not, this is great!

Reply:
They are basically assertions, so they are not supposed to be seen by a human but I want to catch these cases early if something is really screwy

Reply:
Don't `Precondition` calls raise `IllegalStateException`? If that's the case, then it's definitely dev-facing conditions.

Reply:
It looks like [it will throw](https://github.com/Netflix/spectator/blob/5b72bb6f11b4a9f81fd76acd8034ef2bad075376/spectator-api/src/main/java/com/netflix/spectator/impl/Preconditions.java#L45) an `IllegalArgumentException`.

Reply:
`checkArg` raises `IllegalArgumentException`, `checkState` raises `IllegalStateException`

Reply:
Whichever exception it is, it will bubble up to end users, right?